File: Stripe/STPPaymentMethodAlipay.m
//
//  STPPaymentMethodAlipay.m
//  StripeiOS
//
//  Copyright © 2020 Stripe, Inc. All rights reserved.
//

#import "STPPaymentMethodAlipay.h"

#import "NSDictionary+Stripe.h"

@implementation STPPaymentMethodAlipay

@synthesize allResponseFields = _allResponseFields;

#pragma mark - Description

- (NSString *)description {
    NSArray *props = @[
                       // Object
                       [NSString stringWithFormat:@"%@: %p", NSStringFromClass([self class]), self],
                       ];

    return [NSString stringWithFormat:@"<%@>", [props componentsJoinedByString:@"; "]];
}

#pragma mark - STPAPIResponseDecodable

+ (nullable instancetype)decodedObjectFromAPIResponse:(nullable NSDictionary *)response {
    NSDictionary *dict = [response stp_dictionaryByRemovingNulls];
    if (!dict) {
        return nil;
    }
    return [[self alloc] initWithDictionary:dict];
}

- (instancetype)initWithDictionary:(NSDictionary *)dict {
    self = [super init];
    if (self) {
        _allResponseFields = dict.copy;
    }
    return self;
}

@end

File: StripePayments/StripePayments/API Bindings/Models/PaymentMethods/Types/STPPaymentMethodAlipay.swift
//
//  STPPaymentMethodAlipay.swift
//  StripePayments
//
//  Copyright © 2020 Stripe, Inc. All rights reserved.
//

import Foundation

/// Contains details for an Alipay Payment Method.
/// - seealso: https://stripe.com/docs/api/payment_methods/object#payment_method_object-alipay
public class STPPaymentMethodAlipay: NSObject, STPAPIResponseDecodable {
    /// :nodoc:
    private(set) public var allResponseFields: [AnyHashable: Any] = [:]

    // MARK: - Description
    /// :nodoc:
    @objc public override var description: String {
        let props = [
            // Object
            String(format: "%@: %p", NSStringFromClass(STPPaymentMethodAlipay.self), self)
        ]

        return "<\(props.joined(separator: "; "))>"
    }

    // MARK: - STPAPIResponseDecodable
    /// :nodoc:
    public class func decodedObject(fromAPIResponse response: [AnyHashable: Any]?) -> Self? {
        guard let dict = response?.stp_dictionaryByRemovingNulls() else {
            return nil
        }
        return self.init(dictionary: dict)

    }

    required init(
        dictionary dict: [AnyHashable: Any]
    ) {
        super.init()
        allResponseFields = dict
    }
}
